src/Router.cpp

```diff
@@ -110,50 +130,66 @@ void Router::splitTarget(const std::string &target)
 	}
 }
 
-bool Router::pathIsValid(HTTPRequest &request, std::string webRoot)
+bool Router::pathIsValid(HTTPRequest &request, std::string &webRoot)
 {
 	std::string host = request.getHost();
+	std::cout << "pathIsValid Host: " << host << std::endl;
+	std::cout << "pathIsValid WebrRoot: " << webRoot << std::endl;
 	size_t pos = host.find(":");
 	if (pos != std::string::npos)
 	{
 		host = host.substr(0, pos);
 	}
-	std::string pathWithQuery = request.getRequestTarget();
-	std::string path = pathWithQuery.substr(0, pathWithQuery.find("?"));
+	// std::cout << "Host (after : trailing) :" << host << std::endl;
 
-	if (host == "localhost")
-		path = webRoot + path;
-	else
-		path = webRoot + "/" + host + path;
+	// for ease of use during deployment
+	// this if/else allows to reach target with tester or browser
+	// if (host != "localhost")
+	// 	webRoot = host + webRoot;
 
-	std::cout << std::endl << "Path: " << path << std::endl << std::endl;
-	std::cout << "Path: " << path << std::endl;
+	std::cout << std::endl << "pathIsValid Path: " << webRoot << std::endl << std::endl;
+	std::cout << "pathIsValid Path: " << webRoot << std::endl;
 	struct stat buffer;
-	if (stat(path.c_str(), &buffer) != 0)
+	if (stat(webRoot.c_str(), &buffer) != 0)
 	{
+		std::cout << "+-+-+-+-+- does not exist" << std::endl;
+		request.setPath(webRoot);
 		return false;
 	}
 	if (S_ISDIR(buffer.st_mode))
 	{
 		// std::cout << "Path is a directory" << std::endl;
-		if (!path.empty() && path[path.length() - 1] != '/')
+		if (!webRoot.empty() && webRoot[webRoot.length() - 1] != '/')
+		{
+			webRoot += "/";
+		}
+		// IF THIS IS NOT DEFINE IN THE CONFIG FILE, THEN DO WHAT IS BELOW:
+		if (_serverBlock.getIndex().empty())
+		{
+			webRoot += "index.html";
+		}
+		else
 		{
-			path += "/";
+			std::cout << "Index: " << _serverBlock.getIndex()[0] << std::endl;
+			std::string index = _serverBlock.getIndex()[0];
+			webRoot += index;
+			std::cout << "Path: " << webRoot << std::endl;
 		}
-		path += "index.html";
 		// std::cout << "Path: " << path << std::endl;
-		if (stat(path.c_str(), &buffer) != 0)
+		if (stat(webRoot.c_str(), &buffer) != 0)
 		{
+			request.setPath(webRoot);
 			// TODO: decide if we should return a custom error for a directory without an index.html
 			return false;
```

Comment:
We don't want to give extra information about the websites we host. If the index is not present and we don't have that directive that allow to listen the content of the directory, then it's a 404 no?

The friend:

The choice between returning a 404 Not Found error and a 403 Forbidden error when a directory is accessed without an index file and directory listing is disabled depends on the intended message and security policy.

- **404 Not Found**: This error implies that the URL points to a resource that does not exist. Using a 404 error might be more user-friendly or less revealing about the server's structure, as it doesn't explicitly indicate that the directory exists. It simply states that the requested resource is unavailable, which could be interpreted as the directory or page not existing at all.

- **403 Forbidden**: This error explicitly tells the user that the server understood the request but refuses to authorize it. If a directory without an index file is accessed and directory listings are disabled, a 403 error is technically correct because the resource (directory listing or index file) is there but access to it is not allowed. This provides a clear indication that the directory exists but the server's configuration prevents its contents from being listed or an index file from being shown.

Choosing between a 404 and a 403 error often depends on the desired security posture and user experience. For example, if you prefer to minimize information about the server’s directory structure and potential points of vulnerability, a 404 might be the better choice. On the other hand, a 403 error might be used when it's okay to reveal the existence of a directory but not its contents, often used in more secure or private areas of a website.